atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java

```diff
@@ -164,20 +183,189 @@ private long storeAndCheckExpiringByteStreams(int size) throws IOException {
     }
 
     private void verifyLoadingStreams(long id, byte[] bytesToStore, GenericStreamStore<Long> store) throws IOException {
-        byte[] loadedBytes;
-        InputStream stream = txManager.runTaskThrowOnConflict(t -> store.loadStream(t, id));
+        verifyLoadStream(id, bytesToStore, store);
+        verifyLoadStreams(id, bytesToStore, store);
+        verifyLoadStreamAsFile(id, bytesToStore, store);
+    }
 
-        Sha256Hash expectedHash = Sha256Hash.computeHash(bytesToStore);
-        loadedBytes = IOUtils.toByteArray(stream);
-        Assert.assertArrayEquals(bytesToStore, loadedBytes);
-        Assert.assertEquals(expectedHash, Sha256Hash.computeHash(loadedBytes));
+    private void verifyLoadStreamAsFile(long id, byte[] bytesToStore, GenericStreamStore<Long> store) throws IOException {
+        File file = txManager.runTaskThrowOnConflict(t -> store.loadStreamAsFile(t, id));
+        Assert.assertArrayEquals(bytesToStore, FileUtils.readFileToByteArray(file));
+    }
 
+    private void verifyLoadStreams(long id, byte[] bytesToStore, GenericStreamStore<Long> store) throws IOException {
         Map<Long, InputStream> streams = txManager.runTaskThrowOnConflict(t ->
                 store.loadStreams(t, ImmutableSet.of(id)));
+        assertStreamHasBytes(streams.get(id), bytesToStore);
+    }
+
+    private void verifyLoadStream(long id, byte[] bytesToStore, GenericStreamStore<Long> store) throws IOException {
+        InputStream stream = txManager.runTaskThrowOnConflict(t -> store.loadStream(t, id));
+        assertStreamHasBytes(stream, bytesToStore);
+    }
+
+    private void assertStreamHasBytes(InputStream stream, byte[] bytes) throws IOException {
+        byte[] streamAsBytes = IOUtils.toByteArray(stream);
+        Assert.assertArrayEquals(bytes, streamAsBytes);
+    }
+
+    @Test
+    public void testLookupStreamIdsByHash() throws Exception {
+        final byte[] bytes1 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
+        final byte[] bytes2 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
+
+        long id1 = timestampService.getFreshTimestamp();
+        long id2 = timestampService.getFreshTimestamp();
+
+        Random rand = new Random();
```

Comment:
This sounds like a good idea, but I wouldn't want to block this PR on getting it in.